worker/base/worker.go

```diff
@@ -515,3 +535,46 @@ func oneOffProgress(ctx context.Context, id string) func(err error) error {
 		return err
 	}
 }
+
+func detectLayerMediaType(ctx context.Context, cs content.Store, id digest.Digest) (string, error) {
```

Comment:
Can we move this into shared function and avoid duplicate declaration

Reply:
Updated